client/src/main/java/org/apache/rocketmq/client/trace/AsyncTraceDispatcher.java

```diff
@@ -159,6 +161,10 @@ private DefaultMQProducer getAndCreateTraceProducer(RPCHook rpcHook) {
         return traceProducerInstance;
     }
 
+    private String genGroupNameForTrace() {
+        return TraceConstants.GROUP_NAME_PREFIX + "-" + this.group;
```

Comment:
if it can trace when the multi consumers or producers with the same group name?

Reply:
> if it can trace when the multi consumers or producers with the same group name?
Only the first traceProducer can trace, and the other will be start failed. If specify a instanceName for each group, it will create multi client instance.

Reply:
@jonnxu 
Thank you for the reminder. As far as I know, in RocketMQ two producers for different topics can't share the same ProduceGroup, and consumers also. But a consumer and a producer can have the same group. Unfortunately，the trace will failure in such scenes. To fix this problem, client type(consume or produce) will be appended to the trace-group to distinguish the consumer and producer. 

```
 private String genGroupNameForTrace() {
        return TraceConstants.GROUP_NAME_PREFIX + "-" + this.group + "-" + this.type ;
    }
```
When construct` AsyncTraceDispatcher.type` will be set as "PRODUCE" or "CONSUME". 

This PR has been updated.

And my test code as below:

```
public class SameProcessSameTopicTest {
    private static String topic1 = "TopicTest1";
    private static String topic2 = "TopicTest2";

//    private static final String CONSUMER_GROUP1 = "test_c_group1";
//    private static final String CONSUMER_GROUP2 = "test_c_group2";
//
//    private static final String PRODUCER_GROUP1 = "test_p_group1";
//    private static final String PRODUCER_GROUP2 = "test_p_group2";

    private static final String CONSUMER_GROUP1 = "test__group1";
    private static final String CONSUMER_GROUP2 = "test__group2";

    private static final String PRODUCER_GROUP1 = "test__group1";
    private static final String PRODUCER_GROUP2 = "test__group2";

    private static final String NAMR_SRV = "";

    public static void startConsume(final String id, final String group, final String topic) throws MQClientException {
        DefaultMQPushConsumer consumer = new DefaultMQPushConsumer(group, true);
        consumer.setConsumeFromWhere(ConsumeFromWhere.CONSUME_FROM_LAST_OFFSET);
        consumer.subscribe(topic, "*");
        consumer.setNamesrvAddr(NAMR_SRV);
        consumer.setInstanceName(id);
        final AtomicInteger i = new AtomicInteger(0);
        consumer.registerMessageListener(new MessageListenerConcurrently() {
            public ConsumeConcurrentlyStatus consumeMessage(List<MessageExt> msgs, ConsumeConcurrentlyContext context) {
                for (MessageExt msg : msgs) {
                    System.out.println(new Date() + ":" + new String(msg.getBody()) + " consume success! " + msg.getMsgId() + "  " + topic);
                }
                i.getAndAdd(msgs.size());

                System.err.println(id + " Consumed " + i.get() + " messages for " + topic);
                return ConsumeConcurrentlyStatus.CONSUME_SUCCESS;
            }
        });

        consumer.start();
    }

    public static void startProduce(String id, String group, String topic) throws MQClientException {
        DefaultMQProducer producer = new DefaultMQProducer(group,true);
        producer.setNamesrvAddr(NAMR_SRV);
        producer.start();
        producer.setInstanceName(id);
        AtomicInteger c = new AtomicInteger(0);
        for (int i = 0; i < 10 ;) {
            try {
                Message msg = new Message(topic /* Topic */, null /* Tag */,
                        ("trace message " + i).getBytes(RemotingHelper.DEFAULT_CHARSET) /* Message body */
                );
                producer.send(msg);
                Thread.sleep(50);
                i++;
                System.out.println(id + " send " + c.getAndIncrement() + " messages to " + topic);
            } catch (Exception e) {
//                e.printStackTrace();
            }
        }

    }


    public static void main(String[] args) throws InterruptedException, MQClientException {

        Thread t1 = new Thread() {
            @Override
            public void run() {
                try {
                    startConsume("C1", CONSUMER_GROUP1, topic1);
                } catch (MQClientException e) {
                    e.printStackTrace();
                }
            }
        };

        Thread t2 = new Thread() {
            @Override
            public void run() {
                try {
                    startConsume("C2", CONSUMER_GROUP2, topic2);
                } catch (MQClientException e) {
                    e.printStackTrace();
                }
            }
        };


        Thread t3 = new Thread() {
            @Override
            public void run() {
                try {
                    startProduce("P1", PRODUCER_GROUP1, topic1);
                } catch (MQClientException e) {
                    e.printStackTrace();
                }
            }
        };
//
        Thread t4 = new Thread() {
            @Override
            public void run() {
                try {
                    startProduce("P2", PRODUCER_GROUP2, topic2);
                } catch (MQClientException e) {
                    e.printStackTrace();
                }
            }
        };


        t1.start();
        t2.start();
        t3.start();
        t4.start();

        t1.join();
        t2.join();
        t3.join();
        t4.join();

        System.out.printf("Consumer Started.%n");
    }
} 

```